retrofit/src/main/java/retrofit2/OkHttpCall.java

```diff
@@ -58,28 +59,47 @@
   }
 
   @Override public synchronized Request request() {
-    okhttp3.Call call = rawCall;
-    if (call != null) {
-      return call.request();
+    try {
+      return getRawCall().request();
+    } catch (IOException e) {
+      throw new RuntimeException("Unable to create request.", e);
     }
+  }
+
+  @Override public Timeout timeout() {
+    try {
+      return getRawCall().timeout();
+    } catch (IOException e) {
+      throw new RuntimeException("Unable to create call.", e);
+    }
+  }
+
+  /**
+   * Returns the raw call, initializing it if necessary. Throws if initializing the raw call throws,
+   * or has thrown in previous attempts to create it.
+   */
+  private okhttp3.Call getRawCall() throws IOException {
```

Comment:
Needs to be synchronized. Concurrent invocations can return different instances or see partial state.

I hope error-prone catches this because of the `@GuardedBy`...

Reply:
It did. Whew.